### ISSUE TITLE
"Focus indication area" for apparent movement

### ISSUE THREAD

**Issue by JasonAment** [w3c/wcag#1059]
[Code Pen illustrating that a focus indicator does not have to be visible to pass](https://codepen.io/JasonAment/pen/MWwoEvN)

Illustrating that there is no requirement that the pixels that change need to stay in the same position relative to their original position.

The following button passes all of the requirements for the 2.4.11 requirements, but the button shifts position on the page 10px when it receives focus. So while it technically passes, there is no visible indicator.  I can nudge the control a bit and still pass the requirement since I am changing the pixels of the focus indicator.  There is no requirement to indicate that the same pixel has to change color.

This could be updated to say something like "The "focus indication area" is the space occupied at the current x/y position that changes to indicate focus".  

Since there is a [new] paragraph (in 2.4.7) that states 2.4.7 passes if 2.4.11 passes, this failure will ultimately cause both to pass. The added paragraph in 2.4.7 defines the visibility of a 2.4.7 visual focus indicator using 2.4.11. So technically, this "Pass" meets the visibility requirements of 2.4.7.

**Minimum area:** The focus indication area is a 10px border changes from black to white. The total surface area that changes between each state is well over the requirements.
**Focus contrast:** The focus indication area is a 10px changes from black to white. This is a 21:1 contrast ratio difference in states.
**Contrast or thickness:** The focus indication area is a 10px thickness border that changes from black to white on opposing edges. The total surface area that changes between each state is well over the requirements.

**Comment by JasonAment** [comment]
"Focus indication area" is only concerned with the **"number"** of css pixels that change.

**Comment by alastc** [comment]
Hi Jason,

I didn't get the effect in Chrome or Firefox in your pen, but from the description I'm going to assume that the button is supposed to move down and to the right. Whether it does that with actually moving the button or fiddling with border-colors doesn't really matter, it is the pixels that change which count.

In that (fairly odd) case the focus indicator size is the size of the blue + red areas in the second button. Well, it is slightly less due to the top-right & bottom left corners:
<img src="https://user-images.githubusercontent.com/216630/76035065-71300080-5f38-11ea-9e60-c454aa77725d.png" alt="A black button overlaid with another copy of the same button, 10 pixels down and to the right.">

> So technically, this "Pass" meets the visibility requirements of 2.4.7.

It would pass 2.4.7 now, as there is no definition of what visibility means. 

Conceptually it is similar to example 9B on this [test page](https://alastairc.uk/tests/wcag22-examples/focus-more-visible-2.html), where a 2px black outline is added a black button. It just makes the button look slightly larger.

If you look at [several buttons showing the behaviour](https://codepen.io/alastc/pen/poJWYeX), it is odd, not recommended, but it is visible.

Have you come across this approach in the wild? We have other issues talking about the complexity already, I'm struggling to see how to add a location requirement without making it more complex. I'd rather not add complexity for something that people don't do now because it looks odd to everyone. 

Unless I'm mistaken, using a focus-style like that would (until [focus-visible](https://github.com/WICG/focus-visible/blob/master/explainer.md) is more supported and used) show when you click on a button like that, so it would be odd for everyone.

**Comment by alastc** [comment]
Hi @JasonAment,

The working group [discussed this today](https://www.w3.org/2020/06/30-ag-minutes.html#item05) and agreed that:
<hr>
This is not a common pattern that we have seen in the wild, and it seems unlikely that the new success criteria would cause people to try this method. Whilst it is a potential hole, we would rather not complicate the text in order to plug that (quite rare) hole. We will update the understanding document to acknowledge this as a possible outcome.
